src/navi/transform.clj

```diff
@@ -83,12 +83,17 @@
          (map p/transform)
          (into [compose-as]))))
 
+(def byte-array-class (class (byte-array 0)))
+
+(defn binary-data? [x]
```

Comment:
This function needs to be public for the sake of unit testing. Would it make sense to move it to a separate file such as `src/navi/predicates`?